tools/clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -1236,7 +1241,8 @@ llvm::Value *CodeGenFunction::EmitLoadOfScalar(llvm::Value *Addr, bool Volatile,
       EmitCheck(std::make_pair(Check, Kind), "load_invalid_value", StaticArgs,
                 EmitCheckValue(Load));
     }
-  } else if (CGM.getCodeGenOpts().OptimizationLevel > 0)
+  } else if (CGM.getCodeGenOpts().OptimizationLevel > 0 &&
+             ShouldEmitRangeMD(Load, Ty))
```

Comment:
Just as a comment for any additional reviewers. This matches what modern clang does. Basically we can skip inserting value range metadata for bools that are represented as i1 because the valid range of a bool is the full range of an i1.

Reply:
Is bool the only type we would want to emit range information for?  `ShouldEmitRangeMD` won't return true unless hasBooleanRepresentation(Ty) returns true AND has bit size > 1.  It isn't obvious to me that this is correct.

Reply:
That’s a bug in the code I gave @amaiorano. I’ve posted suggestions to correct.